axum-core/src/macros.rs

```diff
@@ -54,7 +54,7 @@ macro_rules! __define_rejection {
             }
 
             /// Get the status code used for this rejection.
-            pub fn status(&self) -> http::StatusCode {
+            pub const fn status(&self) -> http::StatusCode {
```

Comment:
I don't think most of these really make sense. I don't think anyone would built a rejection in a const context to then call this.

I did not go through all of this commit though, there might be some useful ones here, but this lint might also be one to split out of this PR.